credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,317 @@
+   * @param statusListCredential The [VerifiableCredential] representing the status list.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the `statusListCredential`.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck, statusListCred)
+   * ```
+   */
+  public fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredential,
+    statusListCredential: VerifiableCredential
+  ): Boolean {
+    val statusListEntryValue: StatusList2021Entry =
+      StatusList2021Entry.fromJsonObject(credentialToValidate.vcDataModel.credentialStatus.jsonObject)
+
+    val statusListCredStatusPurpose: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[STATUS_PURPOSE] as? String?
+
+    if (statusListEntryValue.statusPurpose == null) {
+      throw IllegalArgumentException("Status purpose in the credential to validate is null")
+    }
+
+    if (statusListCredStatusPurpose == null) {
+      throw IllegalArgumentException("Status purpose in the status list credential is null")
+    }
+
+    if (statusListEntryValue.statusPurpose != statusListCredStatusPurpose) {
+      throw IllegalArgumentException("Status purposes do not match between the credentials")
+    }
+
+    val compressedBitstring: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[ENCODED_LIST] as? String?
+
+    if (compressedBitstring == null || compressedBitstring.isEmpty()) {
+      throw IllegalArgumentException("Compressed bitstring is null or empty")
+    }
+
+    val credentialIndex = statusListEntryValue.statusListIndex
+    val expandedValues: List<String> = bitstringExpansion(compressedBitstring)
+
+    return expandedValues.any { it == credentialIndex }
+  }
+
+
+  /**
+   * Validates if a given credential is part of the status list.
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param httpClient An optional [HttpClient] for fetching the status list credential. If not provided, a default HTTP client will be used.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function fetches the status list credential from a URL present in the `credentialToValidate`.
+   * It supports using either a user-provided `httpClient` or a default client when no client is passed in.
+   * The function then checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the fetched status list credential.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck)
+   * ```
+   */
+  public suspend fun validateCredentialInStatusList(
```

Comment:
We ended up deciding to have all methods be synchronous in the public APIs (at least for now). Could you make is such that this function isn't suspendable?

Reply:
ok updated with runblocking